src/Downloader/Middleware/ExecuteJavascriptMiddleware.php

```diff
@@ -35,7 +35,7 @@ public function __construct(
         private LoggerInterface $logger,
         ?callable $getBrowsershot = null,
     ) {
-        $this->getBrowsershot = $getBrowsershot ?: static fn (string $uri): Browsershot => Browsershot::url($uri);
+        $this->getBrowsershot = $getBrowsershot ?: static fn (string $uri): Browsershot => Browsershot::url($uri)->waitUntilNetworkIdle();
```

Comment:
```
/** @psalm-suppress MixedInferredReturnType, MixedReturnStatement */
```

Add this above line 38 to suppress the Psalm warning. You can check that the error went away locally by running `composer analyze`.

Reply:
Confirmed. Thanks!